vars/buildApp.groovy

```diff
@@ -40,6 +45,11 @@ def call(Map parameters = [:]) {
     currentBuild.displayName = version
     currentBuild.description = "Docker: ${tag}"
 
-    sh "git archive HEAD | docker build -t ${name}:${tag} -"
+    if (!path) {
+        sh "git archive HEAD | docker build -t ${name}:${tag} -"
+    } else {
+        sh "docker build -t ${name}:${tag} ${path}"
+    }
+
```

Comment:
The approach seems sensible to me.

1. A new optional variable called `path` has been introduced.
2. A conditional statement then checks the value of `path` and varies the command run based on it being `NULL` or not.

```
If path is null:
    sh "git archive HEAD | docker build -t ${name}:${tag} -"
```
```
If path is not null:
    sh "docker build -t ${name}:${tag} ${path}"
```